src/encode.rs

```diff
@@ -94,7 +94,7 @@ pub trait Encodable {
 /// Data which can be encoded in a consensus-consistent way
 pub trait Decodable: Sized {
     /// Decode an object with a well-defined format
-    fn consensus_decode<D: io::Read>(d: D) -> Result<Self, Error>;
+    fn consensus_decode<D: io::BufRead>(d: D) -> Result<Self, Error>;
```

Comment:
Maybe mention in the commit message why we are doing this: so we can peek at the prefix and still directly deserialize the commitments.